src/selectors/entities/channel_categories.test.js

```diff
@@ -71,34 +73,98 @@ describe('makeGetChannelsForAllCategories', () => {
                     dmChannel1,
                     gmChannel1,
                 },
+                myMembers: {
+                    [channel1.id]: {},
+                    [channel2.id]: {},
+                    [channel3.id]: {},
+                    [dmChannel1.id]: {},
+                    [gmChannel1.id]: {},
+                },
             },
         },
     };
 
     test('should return channels on the team and DMs/GMs', () => {
-        const getChannelsForAllCategories = Selectors.makeGetChannelsForAllCategories();
+        const getUnsortedUnfilteredChannels = Selectors.makeGetUnsortedUnfilteredChannels();
+
+        expect(getUnsortedUnfilteredChannels(baseState, 'team1')).toMatchObject([channel1, channel2, dmChannel1, gmChannel1]);
+
+        expect(getUnsortedUnfilteredChannels(baseState, 'team2')).toMatchObject([channel3, dmChannel1, gmChannel1]);
+    });
+
+    test('should not return channels which the user is not a member of', () => {
+        const channel4 = {id: 'channel4', team_id: 'team1', delete_at: 0};
+
+        const getUnsortedUnfilteredChannels = Selectors.makeGetUnsortedUnfilteredChannels();
+
+        let state = {
+            entities: {
+                channels: {
+                    channels: {
+                        channel4,
+                    },
+                    myMembers: {},
+                },
+            },
+        };
 
-        expect(getChannelsForAllCategories(state, 'team1')).toMatchObject([channel1, channel2, dmChannel1, gmChannel1]);
+        expect(getUnsortedUnfilteredChannels(state, 'team1')).not.toContain(channel4);
+
+        state = {
+            entities: {
+                channels: {
+                    channels: {
+                        channel4,
+                    },
+                    myMembers: {
+                        [channel4.id]: {},
+                    },
+                },
+            },
+        };
+
+        expect(getUnsortedUnfilteredChannels(state, 'team1')).toContain(channel4);
+    });
+
+    test('should not return deleted channels', () => {
+        const channel = {id: 'channel', team_id: 'team1', delete_at: 0};
+        const deletedChannel = {id: 'deletedChannel', team_id: 'team1', delete_at: 1000};
+
+        const getUnsortedUnfilteredChannels = Selectors.makeGetUnsortedUnfilteredChannels();
+
+        const state = {
+            entities: {
+                channels: {
+                    channels: {
+                        channel,
+                    },
+                    myMembers: {
+                        [channel.id]: {},
```

Comment:
Should this include membership for `deletedChannel`? I think you're testing this same behaviour elsewhere.